### PR TITLE
chore: upgrade to node@18

### PR DESCRIPTION
**Pull request by jye-sf**
## Details
Upgrade pinned node version and CI images to use node@18. Fixes https://github.com/salesforce/lwc/issues/3691

Manually checked and didn't find any changes to the behavior of our scripts. 

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/babel-plugin-component/src/component.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default function ({ types: t }: BabelAPI): Visitor<LwcBabelPluginPass> {
             const hasIdentifier = t.isIdentifier((node as types.ClassDeclaration).id);
             if (hasIdentifier) {
                 statementPath!.insertBefore(node);
-                node = (node as types.ClassDeclaration).id;
+                node = (node as types.ClassDeclaration).id!;
```
Comment (jye-sf) [inline packages/@lwc/babel-plugin-component/src/component.ts:71]: `hasIdentifier` on line 69 checks whether this exists.

File: .circleci/config.yml [hunk 2]
```diff
@@ -268,11 +268,7 @@ jobs:
   test_benchmark_smoke:
     executor: node-browsers
     steps:
-      # TODO [#3691]: remove `apt-get` and `chrome-version` lines when CircleCI issue is fixed
-      # https://github.com/CircleCI-Public/browser-tools-orb/issues/90
-      - run: sudo apt-get update
-      - browser-tools/install-chrome: # required for selenium used by tachometer benchmark smoke tests
-          chrome-version: 116.0.5845.96
+      - browser-tools/install-chrome # required for selenium used by tachometer benchmark smoke tests
```
Comment (jye-sf) [inline .circleci/config.yml:271]: No longer needed, and needs to be removed so we can use the correct chrome 117 version.

File: .circleci/config.yml [hunk 3]
```diff
@@ -268,11 +268,7 @@ jobs:
   test_benchmark_smoke:
     executor: node-browsers
     steps:
-      # TODO [#3691]: remove `apt-get` and `chrome-version` lines when CircleCI issue is fixed
-      # https://github.com/CircleCI-Public/browser-tools-orb/issues/90
-      - run: sudo apt-get update
-      - browser-tools/install-chrome: # required for selenium used by tachometer benchmark smoke tests
-          chrome-version: 116.0.5845.96
```
Comment (jye-sf) [inline .circleci/config.yml:275]: No longer needed, and needs to be removed so we can use the correct chrome 117 version.

Comment (nolanlawson) [reply .circleci/config.yml:275]: Could you put "Fixes #3691" in the PR description?

**Review by nolanlawson — APPROVED** [review]

File: packages/@lwc/babel-plugin-component/src/component.ts (hunk 1 above)
Comment (jye-sf) [reply packages/@lwc/babel-plugin-component/src/component.ts:71]: Yes. Without this change, we get this error when trying to build:
```
 nx run @lwc/babel-plugin-component:build
$ rollup --config ../../../scripts/rollup/rollup.config.js
       
       /Users/jye/dev/lwc/packages/@lwc/babel-plugin-component/src/index.ts → dist/index.js, dist/index.cjs.js...
       [!] (plugin typescript) RollupError: @rollup/plugin-typescript TS2322: Type 'Identifier | null | undefined' is not assignable to type 'ClassDeclaration | FunctionDeclaration | Expression'.
         Type 'undefined' is not assignable to type 'ClassDeclaration | FunctionDeclaration | Expression'.
       src/component.ts (71:17)
       
       71                 node = (node as types.ClassDeclaration).id;
```

Comment (nolanlawson) [reply packages/@lwc/babel-plugin-component/src/component.ts:71]: Bizarre. No problem though!

**Review by ravijayaramappa — APPROVED** [review]

**Pull request merged**